packages/session-replay-browser/src/remote-config-fetch.ts

```diff
@@ -31,15 +32,53 @@ export class SessionReplayRemoteConfigFetch implements AmplitudeSessionReplayRem
     this.sessionIDBStore = sessionIDBStore;
   }
 
+  /**
+   * Once targeting has been evaluated to positive for the session,
+   * it should not be reevaluated. A sessionTargetingMatch of true should
+   * stay true for the remainder of the session.
+   */
+  evaluateTargeting = async (targetingParams: Omit<TargetingParameters, 'flag'>) => {
+    // First check memory for existing true decision
+    if (this.sessionTargetingMatch === true) {
+      return;
+    }
+
+    // Then check IndexedDB for existing true decision
+    const idbTargetingMatch = await this.sessionIDBStore.getTargetingMatchForSession(targetingParams.sessionId);
+    if (idbTargetingMatch === true) {
+      this.sessionTargetingMatch = true;
+      return;
+    }
+
+    // Finally evaluate targeting if previous two checks were false or undefined
+    try {
+      const targetingConfig = await this.getTargetingConfig(targetingParams.sessionId);
+      if (targetingConfig && Object.keys(targetingConfig).length > 0) {
+        const targetingResult = evaluateTargeting({ ...targetingParams, flag: targetingConfig });
+        this.sessionTargetingMatch =
+          this.sessionTargetingMatch === false && targetingResult.sr_targeting_config.key === 'on';
+      } else {
+        // If the targeting config is undefined or an empty object,
+        // assume the response was valid but no conditions were set,
+        // so all users match targeting
+        this.sessionTargetingMatch = true;
```

Comment:
How come we assume true here if config is undefined?

Reply:
We want to provide backwards compatibility to the existing functionality. So that means that some customers won't have any targeting config defined (especially when we are in the beta phase). In that case, we'll set this to true, so that we can know to start sending events to the backend